alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDeleteWorkflowAction.kt

```diff
@@ -193,24 +206,36 @@ class TransportDeleteWorkflowAction @Inject constructor(
             }
         }
 
-        private suspend fun deleteMonitors(monitorIds: List<String>, refreshPolicy: RefreshPolicy) {
-            if (monitorIds.isEmpty())
-                return
+        /**
+         * Tries to delete the given list of the monitors. Return value contains all the monitorIds which deletion failed
+         * @param monitorIds list of monitor ids to be deleted
+         * @param refreshPolicy
+         * @return list of the monitors that were not deleted
+         */
+        private suspend fun tryDeletingMonitors(monitors: List<Monitor>, refreshPolicy: RefreshPolicy): List<String> {
+            val nonDeletedMonitorIds = mutableListOf<String>()
 
-            for (monitorId in monitorIds) {
-                val deleteRequest = DeleteMonitorRequest(monitorId, refreshPolicy)
-                val searchResponse: DeleteMonitorResponse = client.suspendUntil {
-                    AlertingPluginInterface.deleteMonitor(this as NodeClient, deleteRequest, it)
+            for (monitor in monitors) {
+                val monitorId = monitor.id
+                try {
+                    val monitorIsWorkflowDelegate = DeleteMonitorService.monitorIsWorkflowDelegate(monitorId)
+                    if (monitorIsWorkflowDelegate) {
+                        nonDeletedMonitorIds.add(monitorId)
+                    }
```

Comment:
Why is this needed? We already check if this monitor is not part of any other workflow.

Reply:
You are right. 
We already filtered the monitors that can be deleted(we can delete only the monitors belonging to a given workflow - if the monitor is shared between multiple workflows - those ones will be skipped)